packages/components/addon/components/hds/tabs/index.js

```diff
@@ -0,0 +1,90 @@
+import Component from '@glimmer/component';
+import { A } from '@ember/array';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class HdsTabsIndexComponent extends Component {
+  // UNCOMMENT THIS IF YOU NEED A CONSTRUCTOR
+  // constructor() {
+  //   super(...arguments);
+  //   // ADD YOUR ASSERTIONS HERE
+  // }
+
+  @tracked tabNodes = A([]);
+  @tracked tabIds = A([]);
+  @tracked panelNodes = A([]);
+  @tracked panelIds = A([]);
+  @tracked selectedIndex = 0; // isSelected, 0 = default
+
+  @action
+  didInsertTab(tabId, isSelected, element) {
+    this.tabNodes = A([...this.tabNodes, element]);
+    this.tabIds = A([...this.tabIds, tabId]);
+
+    if (isSelected) {
+      this.setSelectedTabIndex(this.tabIds.length - 1);
+    }
+  }
+
+  @action
+  didInsertPanel(panelId, element) {
+    this.panelNodes = A([...this.panelNodes, element]);
+    this.panelIds = A([...this.panelIds, panelId]);
+  }
+
+  @action
+  handleClick(index, e) {
+    this.setSelected(index, e);
+  }
+
+  @action
+  handleKeyUp(index, e) {
+    const leftArrow = 37;
+    const rightArrow = 39;
+    const downArrow = 40;
+
+    if (e.keyCode === rightArrow) {
+      if (index < this.tabIds.length - 1) {
+        index++;
+        this.setSelected(index, e);
+      }
+    } else if (e.keyCode === leftArrow) {
+      if (index > 0) {
+        index--;
+        this.setSelected(index, e);
+      }
+    } else if (e.keyCode === downArrow) {
+      e.preventDefault();
+      this.setSelectedPanelFocus(index, e);
```

Comment:
I added focus for the tabPanel when using the arrow key to follow this example linked from the CRD: https://inclusive-components.design/tabbed-interfaces/ (CodePen of example from article: https://codepen.io/heydon/pen/veeaEa)

Would like @MelSumner feedback on if this is the desired behavior or if I should do something different.

Reply:
Consider also this: as a user I navigate `left/right` with the arrows, and see the focus move; then I click the `down` arrow and the focus moves down to the panel; now if I click the `up` arrow... nothing happens (I would have expect the focus to move back to the tab) and if I click `left/right/down/up` nothing happens anymore, I'm kinda stuck.

Reply:
@didoo, I had the same experience and thought. I believe we're going to discuss this with @MelSumner next week before making a decision. Not sure I want to hold up the ticket for that, or if it's something that can be addressed later. Thoughts?

Reply:
since this is an intermediate ticket, I'm OK to wait next week and take a decision later

Reply:
I followed what was done in the article linked from the CRD, as I said, to start. I would like feedback from @MelSumner before making further changes.

Reply:
(about the up arrow doing nothing, to me it doesn't make clear sense either but I followed what was in the linked article since I don't know what blind users expect so I relied upon it to start. If @MelSumner says we should add additional functionality like that though then I'll add it)

Reply:
I added the cycling behavior for left and right arrows now since 2 out of the 3 linked examples from the CRD including the example from WAI Aria included it.

Reply:
Still would like more feedback from @MelSumner when she returns regarding up arrow behavior since none of the linked examples had that behavior.